src/PMGDIterators.h

```diff
@@ -132,4 +134,152 @@ namespace VDMS {
 
         PMGD::Node *ref() { return &**_neighb_i; }
     };
+
+    class PMGDQueryHandler::NodeEdgeIteratorImpl : public PMGD::EdgeIteratorImplIntf
+    {
+        /// Reference to expression to evaluate
+        const SearchExpression _expr;
+        const size_t _num_predicates;
+
+        ReusableNodeIterator *_src_ni;
+        ReusableNodeIterator *_dest_ni;
+
+        // In order to check if the other end of an edge is in the nodes
+        // covered by the dest_ni, it is best to store those nodes in an
+        // easily searchable data structure, which a list inside ReusableNodeIterator
+        // is not. Besides, it doesn't make sense to expose that list here.
+        std::unordered_set<PMGD::Node *> _dest_nodes;
+
+        std::size_t _pred_start;
+        PMGD::Direction _dir;
+        bool _check_dest;
+
+        PMGD::EdgeIterator *_edge_it;
+
+        bool _next();
+        bool check_predicates();
+
+        PMGD::EdgeIterator return_iterator()
+        {
+            _dir = PMGD::Direction::Outgoing;
+            if (_src_ni == NULL) {
+                if (_dest_ni == NULL)
+                    _pred_start = 1;
+                else {
+                    _dir = PMGD::Direction::Incoming;
+                    _src_ni = _dest_ni;
+                    _dest_ni = NULL;
+                }
+            }
+            return (_src_ni == NULL || !bool(*_src_ni)) ? _expr.db().get_edges(_expr.tag(),
```

Comment:
split this into if/elses.

Reply:
even if this works, !bool(*_src_ni) will never be empty because of how the code is right now, but we should change in the future because we want to continue with the transaction even if some querynode did not find anything. 

return null if the src iterators is empty.